src/Tvheadend.cpp

```diff
@@ -1917,9 +1907,64 @@ void CTvheadend::ParseRecordingAddOrUpdate ( htsmsg_t *msg, bool bAdd )
   rec.SetStop(stop);
 
   /* Channel is optional, it may not exist anymore */
-  if (!htsmsg_get_u32(msg, "channel", &channel))
+  if (!htsmsg_get_u32(msg, "channel", &channel) && channel > 0)
+  {
+    /* Channel Id */
     rec.SetChannel(channel);
 
+    auto cit = m_channels.find(rec.GetChannel());
+    if (cit != m_channels.cend())
+    {
+      /* Channel type */
+      rec.SetChannelType(cit->second.GetType());
+
+      /* Channel name */
+      rec.SetChannelName(cit->second.GetName());
+    }
+  }
+
+  /* Channel type fallback (in case channel was deleted) */
+  if (!rec.GetChannelType() && m_conn.GetProtocol() >= 25)
+  {
+    htsmsg_t *files, *streams;
+    if ((files = htsmsg_get_list(msg, "files")) != NULL)
+    {
+      htsmsg_field_t *file, *stream;
+      uint32_t hasAudio = 0, hasVideo  = 0, u32;
+
+      HTSMSG_FOREACH(file, files) // Loop through all files
+      {
+        if (file->hmf_type != HMF_MAP)
+          continue;
+
+        if ((streams = htsmsg_get_list(&file->hmf_msg, "info")) != NULL)
+        {
+          HTSMSG_FOREACH(stream, streams) // Loop through all streams
+          {
+            if (stream->hmf_type != HMF_MAP)
+              continue;
+
+            if (!htsmsg_get_u32(&stream->hmf_msg, "audio_type", &u32)) // Only present for audio streams
+              hasAudio = 1;
+            if (!htsmsg_get_u32(&stream->hmf_msg, "aspect_num", &u32)) // Only present for video streams
+              hasVideo = 1;
+            if (hasAudio && hasVideo) // No need to parse the rest
+              break;
+          }
+        }
+      }
+      rec.SetChannelType(hasVideo ? PVR_RECORDING_CHANNEL_TYPE_TV :
+          (hasAudio ? PVR_RECORDING_CHANNEL_TYPE_RADIO : PVR_RECORDING_CHANNEL_TYPE_UNKNOWN));
+    }
+  }
+
+  /* Channel name fallback (in case channel was deleted) */
+  if (rec.GetChannelName().empty()  && m_conn.GetProtocol() >= 25)
```

Comment:
Please remove one of the two spaces between <code>empty()</code> and <code>&& m_conn</code>.